### DISCUSSION TITLE
Websockets API: Configuration settings or callbacks to force WSS

### DISCUSSION THREAD

**Discussion by actbinary** [emscripten-core/emscripten#19072] [Q&A] [answered]
Is there a way to hook into the emscripten API before a websocket connection is made? At the stage where it has created the URL and before calling the Javascript API to attempt the websocket connection?

Or an option to force all connections via websockets to use WSS instead of WS (ie over https instead of http)

I've just been editing the javascript output manually where it checks the url and changing the prefix to wss://

This is for an app written in C/C++ using sockets [(ENet)](https://github.com/mnaamani/enet-emscripten) for a simple chat server/client (see the tutorial for Enet) . Both compile cleanly via emscripten. Client works in browsers and server via Node.js They connect nicely over http and works as intended without any changes to the enet library code. 

The problems begin when hosting it on a https website. The C/C++ code that works with raw sockets have no idea about SSL and should not need it. Browsers can use https and the node server can be put behind a reverse proxy to terminate the https connection to it. Browsers will not allow unsecured connections from a https page. I need a way to force it to always use wss when opening a websocket connection via the browser.

Ideally I would also like a way to get a hold of any websocket emscripten creates, look at the http request headers it handles. In my use case I cant use functions like emscripten_websocket_set_onopen_callback() because I am not opening the raw sockets myself, it done via the Enet library.

Just a few pages of the source that seemed relevant. 
https://github.com/emscripten-core/emscripten/blob/main/system/lib/websocket/websocket_to_posix_socket.c
https://github.com/emscripten-core/emscripten/blob/main/tools/websocket_to_posix_proxy/src/main.cpp

**Comment by sbc100** [comment, marked as answer]
Isn't this what the `WEBSOCKET_URL` setting is for? 

https://github.com/emscripten-core/emscripten/blob/055c6e881299691248b6347c0642baf8a4bae03a/src/settings.js#L367-L373

**Reply by actbinary** [reply]
That works. Did not know that existed. Thank you.

**Reply by sbc100** [reply]
Great!
